Tests/Common/Communication/EventStorageTests.swift

```diff
@@ -0,0 +1,165 @@
+@testable import CioInternalCommon
+import SharedTests
+import XCTest
+
+class EventStorageTest: UnitTest {
+    var eventStorageManager: EventStorageManager!
+
+    override func setUp() {
+        super.setUp()
+        if let tempDirectory = createTemporaryDirectory() {
+            eventStorageManager = EventStorageManager(logger: log, jsonAdapter: jsonAdapter)
+            Task {
+                await eventStorageManager.updateBaseDirectory(baseDirectory: tempDirectory)
+            }
+        }
+    }
+
+    override func tearDown() {
+        eventStorageManager = nil
+        super.tearDown()
+    }
+
+    // MARK: - Helper Methods
+
+    func createTemporaryDirectory() -> URL? {
+        let temporaryDirectoryURL = URL(fileURLWithPath: NSTemporaryDirectory(), isDirectory: true)
+        let uniqueDirectoryURL = temporaryDirectoryURL.appendingPathComponent(UUID().uuidString)
+        do {
+            try FileManager.default.createDirectory(at: uniqueDirectoryURL, withIntermediateDirectories: true)
+            return uniqueDirectoryURL
+        } catch {
+            print("Failed to create temporary directory: \(error)")
+            return nil
+        }
+    }
+
+    // MARK: - Event Storage Tests
+
+    func test_storeEvent_givenValidEvent_expectFileExists() async throws {
+        let event = ProfileIdentifiedEvent(identifier: "testID")
+        try await eventStorageManager.store(event: event)
+
+        let eventFileURL = await eventStorageManager.baseDirectory
+            .appendingPathComponent(ProfileIdentifiedEvent.key)
+            .appendingPathComponent("\(event.storageId).json")
+        let fileExists = FileManager.default.fileExists(atPath: eventFileURL.path)
```

Comment:
while I understand what we are validating, this is a very fragile test as it would break even if the storage behavior is not breaking. i.e if the internal implementation changes which path it stores the data, the users of the storage won't be impacted, yet the test will break. Can we do something to avoid needed to replicate or know the internal implementation in order to validate the behavior?

Edit: one idea is to have multiple instances of the event storage, and see if writing using one instance and reading the value through the other instance will give you the correct data? this way you validated the persistency, but not the internal details

Reply:
that's a fair point, updating the test